base/test-standard/src/test/java/org/openscience/cdk/aromaticity/AromaticTypeMatcherTest.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) 2024 John Mayfield
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+package org.openscience.cdk.aromaticity;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.openscience.cdk.exception.InvalidSmilesException;
+import org.openscience.cdk.graph.Cycles;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IChemObjectBuilder;
+import org.openscience.cdk.silent.SilentChemObjectBuilder;
+import org.openscience.cdk.smiles.SmilesParser;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Set;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class AromaticTypeMatcherTest {
+
+    private final Set<AromaticType> REMAINING = Collections.synchronizedSet(EnumSet.allOf(AromaticType.class));
+
+    void assertAtomType(String smi, AromaticType expected) throws InvalidSmilesException {
+        IChemObjectBuilder bldr = SilentChemObjectBuilder.getInstance();
+        SmilesParser sp = new SmilesParser(bldr);
+        IAtomContainer mol = sp.parseSmiles(smi);
+        Cycles.markRingAtomsAndBonds(mol);
+        AromaticType type = AromaticTypeMatcher.getType(mol.getAtom(0));
+        Assertions.assertEquals(expected, type);
+        REMAINING.remove(type);
+    }
+
+    @AfterAll
+    void ensureAllTypesTested() {
```

Comment:
@johnmay BTW this worked quite well here to check all the atom types were covered.